src/App.js

```diff
@@ -8,51 +8,66 @@ class App extends React.Component {
     super(props)
 
     this.state={
-      list:List
+      list:List,
+      newTaskName:''
     }
-
-    this.add=this.add.bind(this)
-    this.clear=this.clear.bind(this)
-    this.edit=this.edit.bind(this)
   }
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
   render() {
     return (
       <div>
-        <h2>Welcome to your Todo App!</h2>
         <TodoList
           list={this.state.list}
           add={this.add}
           clear={this.clear}
           edit={this.edit}
+          formChange={this.formChange}
+          newTaskName={this.state.newTaskName}
         />
       </div>
     );
   }
-
-  add(e){
+  
+  componentDidMount(){
+    if(localStorage.getItem('tasks'))
+      this.setState({list:JSON.parse(localStorage.getItem('tasks'))})
+  }
+  componentDidUpdate(){
+    localStorage.setItem('tasks',JSON.stringify(this.state.list))
+  }
+  
+  add=e=>{
+    e.preventDefault();
+    
     let temp=this.state.list.map(i=>i)
-    temp.push(e)
-    localStorage.setItem('tasks',JSON.stringify(temp))
-    this.setState({list:temp})
+    
+    temp.push({
+      task:this.state.newTaskName,
+      id:Math.round(Math.random()*100000 + 1),
+      completed:false
+    })
+    this.setState({list:temp,newTaskName:''})
   }
 
-  componentDidMount(){
-    // if(localStorage.getItem('tasks'))
-    //   this.setState({list:JSON.parse(localStorage.getItem('tasks'))})
-  }
   
-  clear(){
+  clear=()=>{
     let temp=this.state.list.map(i=>i).filter(i=>i.completed===false)
     this.setState({list:temp})
   }
 
-  edit(item){
+  edit=item=>{
     let temp=this.state.list.map(i=>(i.id!==item)?i:{task:i.task,id:i.id,completed:!i.completed})
     this.setState({list:temp})
   }
+
+  formChange=e=>{
+    if(e.key==='Enter')
+        this.handleSubmit()
+    else
+        this.setState({newTaskName:e.target.value})
+  }
```

Comment:
You could actually get more millage out of this funciton definition. This can be made to be used on multiple inputs on the same form or across the application if you use bracket notation and assign a `name` attribute to whatever it the field in state you want to modify is. You can then set the state with `{[e.target.name]: e.target.value}` and that will become a more dynamic input handler. 

I am also fairly sure you dont need the logic here to determine whether the key that was pressed was enter or not, this is handled by the `onSubmit` synthetic event in React.